lib/rabbitmqctl.ex

```diff
@@ -229,4 +233,18 @@ defmodule RabbitMQCtl do
     :net_kernel.stop
     exit({:shutdown, code})
   end
+
+  @spec with_distribution(options(), (() -> command_result())) :: command_result()
+  defp with_distribution(options, code) do
+    # Tries to start net_kernel distribution, and calls `code`
+    # function on success. Otherswise returns error suitable for
+    # handle_shutdown/0.
+    case Distribution.start(options) do
+      :ok ->
```

Comment:
`Distribution.start/1` returns a tuple of 2 (because `net_kernel:start/1` does).